app/services/flow/base_step.rb

```diff
@@ -9,12 +9,15 @@ def initialize(flow, name)
     end
 
     def base_call
-      form_response = form_submit
-      unless form_response.success?
-        flow_session[:error_message] = form_response.errors
-        return form_response
+      @form_response = form_submit
+      unless @form_response.success?
+        flow_session[:error_message] = @form_response.errors
+        return @form_response
       end
-      call
+      result = call
+      return @form_response if result.nil? || !result.respond_to?(:success?)
+      FormResponse.new(success: result.success?, errors: result.errors,
+                       extra: result.extra.merge(@form_response.extra))
```

Comment:
1. I don't think we need this to be an instance variable here, `@form_response` is not used outside of the method

2. How does this form conversion combine with the check at the end of BaseFlow? Is it fine? https://github.com/18F/identity-idp/blob/master/app/services/flow/base_flow.rb#L39-L50